meadow-client/src/Semantics.purs

```diff
@@ -0,0 +1,1067 @@
+showValue (ValueFromChoice idChoice value) = joinWith "" ["ValueFromChoice ",
+                                                          showIdChoice idChoice,
+                                                          " (", showValue value, ")"]
+showValue (ValueFromOracle idOracle value) = joinWith "" ["ValueFromOracle ",
+                                                          toString idOracle,
+                                                          " (", showValue value, ")"]
+
+instance showValueI :: Show Value where
+  show = showValue
+
+data Observation = BelowTimeout Timeout |
+                   AndObs Observation Observation |
+                   OrObs Observation Observation |
+                   NotObs Observation |
+                   ChoseThis IdChoice Choice |
+                   ChoseSomething IdChoice |
+                   ValueGE Value Value |
+                   ValueGT Value Value |
+                   ValueLT Value Value |
+                   ValueLE Value Value |
+                   ValueEQ Value Value |
+                   TrueObs |
+                   FalseObs
+derive instance eqObservation :: Eq Observation  
+derive instance ordObservation :: Ord Observation  
+
+showObservation :: Observation -> String
+showObservation (BelowTimeout timeout) =
+  joinWith " " ["BelowTimeout", toString timeout]
+showObservation (AndObs observation1 observation2) =
+  joinWith "" ["AndObs (",
+               showObservation observation1, ") (",
+               showObservation observation2, ")"]
+showObservation (OrObs observation1 observation2) =
+  joinWith "" ["OrObs (",
+               showObservation observation1, ") (",
+               showObservation observation2, ")"]
+showObservation (NotObs observation) =
+  joinWith "" ["NotObs (", showObservation observation, ")"]
+showObservation (ChoseThis idChoice choice) =
+  joinWith " " ["ChoseThis",
+                showIdChoice idChoice,
+                toString choice]
+showObservation (ChoseSomething idChoice) =
+  joinWith " " ["ChoseSomething", showIdChoice idChoice]
+showObservation (ValueGE value1 value2) =
+  joinWith "" ["ValueGE (",
+               showValue value1, ") (",
+               showValue value2, ")"]
+showObservation (ValueGT value1 value2) =
+  joinWith "" ["ValueGT (",
+               showValue value1, ") (",
+               showValue value2, ")"]
+showObservation (ValueLT value1 value2) =
+  joinWith "" ["ValueLT (",
+               showValue value1, ") (",
+               showValue value2, ")"]
+showObservation (ValueLE value1 value2) =
+  joinWith "" ["ValueLE (",
+               showValue value1, ") (",
+               showValue value2, ")"]
+showObservation (ValueEQ value1 value2) =
+  joinWith "" ["ValueEQ (",
+               showValue value1, ") (",
+               showValue value2, ")"]
+showObservation TrueObs = "TrueObs"
+showObservation FalseObs = "FalseObs"
```

Comment:
Me again, back with those pamphlets I mentioned.

Reply:
No, hm, sorry, we are not intersted...